### PR TITLE
Fix  Formatting of Copy/Paste of NR Logs

### LINKED ISSUE #1444: Cleaner Formatting of Copy/Paste of NR Logs
### Current Behavior

When copy/pasting the raw NR logs from FlowForge, I expect formatting to be consistent with the structure shown in FlowForge:

Screenshot 2022-12-15 at 14 44 08

Instead, it pastes as:

```
13/12/2022 14:29:34
[info]
Welcome to Node-RED
===================
13/12/2022 14:29:34
[info]
Node-RED version: v3.0.2
13/12/2022 14:29:34
[info]
Node.js  version: v16.13.2
13/12/2022 14:29:34
[info]
Darwin 22.1.0 arm64 LE
13/12/2022 14:29:34
[info]
Loading palette nodes
13/12/2022 14:29:35
[warn]
------------------------------------------------------
13/12/2022 14:29:35
[warn]
[@flowforge/nr-file-nodes/file] 'file in' already registered by module node-red
13/12/2022 14:29:35
[warn]
[@flowforge/nr-project-nodes/project-link] Error: Project Link nodes cannot be loaded outside of flowforge EE environment (line:6)
13/12/2022 14:29:35
[warn]
------------------------------------------------------
13/12/2022 14:29:35
[info]
Settings file  : /Users/joepavitt/Documents/flowforge/development/flowforge/flowforge/var/projects/9b65d15b-4ab1-4b97-97fb-9ac5c0d228a3/settings.js
13/12/2022 14:29:35
[info]
Context store  : 'default' [module=memory]
13/12/2022 14:29:35
[info]
Server now running at http://127.0.0.1:12080/
13/12/2022 14:29:35
[warn]
Encrypted credentials not found
13/12/2022 14:29:35
[info]
Starting flows
13/12/2022 14:29:35
[info]
Started flows
15/12/2022 14:41:37
[]
15 Dec 14:41:37 - [red] Uncaught Exception:
15/12/2022 14:41:37
[error]
HTTPError: Response code 401 (Unauthorized)
    at Request.<anonymous> (/Users/joepavitt/Documents/flowforge/development/flowforge/flowforge/node_modules/got/dist/source/as-promise/index.js:118:42)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
15/12/2022 14:41:37
[system]
Node-RED exited rc=1 signal=null
15/12/2022 14:41:37
[system]
Starting Node-RED
15/12/2022 14:41:37
[error]
Failed to start server:
15/12/2022 14:41:37
[error]
HTTPError: Response code 401 (Unauthorized)
    at Request.<anonymous> (/Users/joepavitt/Documents/flowforge/development/flowforge/flowforge/node_modules/got/dist/source/as-promise/index.js:118:42)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### Expected Behavior

The pasted content should format nicely, with the date/type/info all on the same line

### Steps To Reproduce

Copy & paste the contents from the `/project/<project-id>/logs` window into a text editor or Slack.

### Environment

- FlowForge version: `main`
- Node.js version:
- npm version:
- Platform/OS:
- Browser:

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Description

https://github.com/FlowFuse/flowfuse/assets/110285294/7f39c12b-3734-494d-88e9-c2fa0a594f29

Screenshot_2023-11-21_22-28-08

in this change i have use a single span tag to display each item's content. It concatenates the properties like item.date, item.level, item.msg, etc. without creating separate p or div  tag  for each entry. It also includes a br tag to add line breaks between entries, except for the last one. it ensure NR log format will not break and there will be no extra line or space between two rows of log

## Related Issue(s)
Closes #1444 

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by sumitshinde-84** [comment]
Hi @joepavitt, could you review this pull request? If any changes are required, please let me know.

**Comment by sumitshinde-84** [comment]
If we want the format to remain as it is, then all entries should be concatenated into a single string. I think that would make it possible.

**Comment by joepavitt** [comment]
> If we want the format to remain as it is,

Not sure I follow here what has changed re: the format?

**Comment by joepavitt** [comment]
Anything that could be done about the extra line between each copied line do yo think?

**Comment by sumitshinde-84** [comment]
> > If we want the format to remain as it is,
> 
> Not sure I follow here what has changed re: the format?

In this change, rather than using a div to log each value. I have used the p tag to log each value within span (display by default inline), ensuring the values will not appear as different rows when we to try to copy.

**Comment by sumitshinde-84** [comment]
> Anything that could be done about the extra line between each copied line do yo think?

I believe the issue will be resolved if we use a single string for the entire log details

**Comment by sumitshinde-84** [comment]
Hi @joepavitt, could you provide your thoughts on what I have discussed in the #2674 issue section?"

**Comment by knolleary** [comment]
@sumitshinde-84 for future PRs, please add a written description to help describe the changes being proposed. The video alone is hard to see what exactly is being changed.

I've had a quick play and can't find an immediate solution to the extra newlines that are appearing between the log lines. It's a definite improvement over what was there before - but would be great to solve this last bit.

**Comment by sumitshinde-84** [comment]
@knolleary  Thanks for reviewing the PR. I completely understand the need for a written description. I'll make sure to include that in the future PRs for better clarity on the proposed changes. I'll continue investigating to find a final solution which will ensure perfect format of NR log for copy-paste.

**Comment by sumitshinde-84** [comment]
Hi @knolleary, I've updated the PR with an improved solution. It ensures that the format won't break when copying and pasting. Also, there won't be any extra lines or spaces between two NR logs. I've added a new video, a screenshot, and a description of the changes. Take a look and let me know if anything else needs adjusting

**Comment by knolleary** [comment]
Thanks @sumitshinde-84 

testing it locally, I think some of the whitespacing is a bit off - but I can't put my finger on it exactly. I'll give this a proper look later today.

**Comment by ZJvandeWeg** [comment]
@knolleary @sumitshinde-84 Any update on this PR?

**Comment by sumitshinde-84** [comment]
> @knolleary @sumitshinde-84 Any update on this PR?

I am waiting for @knolleary's review. Maybe there are whitespaces somewhere. After receiving the review, I will fix them based on suggestions.

File: frontend/src/pages/instance/components/InstanceLogs.vue [hunk 1]
```diff
@@ -10,17 +10,27 @@
...
+                data-el="instance-log-row"
+            >
+                <template v-if="instance.ha?.replicas !== undefined">
+                    [{{ item.src }}]
+                </template>
+                <span>{{ item.date }}</span>
+                <span>{{ "  " }}</span>
+                <span>[{{ item.level }}]</span>
+                <span v-if="item.level !=='system'">{{ "   " }}</span>
+                <span v-else>{{ " " }}</span>
```
Comment (knolleary) [inline frontend/src/pages/instance/components/InstanceLogs.vue:30]:
```suggestion
                <span>{{ `[${item.level || ''}]`.padEnd(10, ' ') }}</span>
```

This fixes the whitespace anomaly I saw in the logs.

**Review by knolleary — APPROVED** [review]
Approved pending one minor change to the whitepsace

**Pull request merged**